.github/workflows/release.yaml

```
@@ -26,6 +26,13 @@ on:
 jobs:
   publish-to-pypi:
     runs-on: ubuntu-22.04
+    permissions:
```

I think we need to declare this as well if we have PyPI trust this workflow + environment.

```suggestion
    environment: release
    permissions:
```

I'm not confident this is a practice we want in general or not, but its something we can easily adjust over time. This project is piloting use of trusted workflows as it is, so lets try this out!